### DISCUSSION TITLE
Unity 6 Support

### DISCUSSION THREAD

**Discussion by francescoStrada** [MixedRealityToolkit/MixedRealityToolkit-Unity#790] [General]
Hello! I am updating (and porting) an application that should run on Quest 3 and Hololens 2. A key feature I need is anchor persistence. Users can design their environment by placing objects in the desired position of the environment, and in subsequent sessions, these objects should be loaded in the previously decided position.

Anchors' persistence on the Hololens was never a problem as I relied on the ARAnchorStore. The problem now is on the Quest 3, which supports persistence only through the Meta XR packages. Unfortunately, I have had problems configuring a project which featured both the MRTK3 and the Meta XR packages. They seem to conflict on the interaction part and generally, having two packages to manage and possible source of problems does not seem a smart solution. 

To this end, the most important features (for example, passthrough) I need from the Quest 3 can be enabled using the AR Foundation package, which (up to now) nicely coexists with MRTK3. 

Only one feature is missing, which is Anchors Persistence, which has recently been added to ARFoundation 6.0, but unfortunately, this version relies on Unity 6.0.

So my question is, are there plans on the MRTK side to eventually upgrade to Unity 6.0? If so, is there a roadmap?

Thnak you for your time, cheers!

**Comment by AMollis** [comment]
Hi @francescoStrada ,  

The MRTK3 contributors are currently working on adding XRI3 support (the support needed for Unity 6).  You can follow along with progress by track the following issues:

- https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/issues/645
- https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/issues/778

If you desire, you can start testing the changes by consuming MRTK3 from this branch

- https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/tree/feature/XRI3

**Comment by YimingSun60** [comment]
Hi, could you tell more about ARAnchorStore? I cannot find it on Google. I'm looking for a way to achieve anchor's persistence on HoloLens2 but ARfoundation do not support save/load anchors on HoloLens, and the Azure Spatial Anchor has retired.

**Reply by BertrandDeSaintBezier** [reply]
I don't think this is at all related to the Unity 6 topic, but if it can help, I recently was working through the same issue and spent a long time juggling different APIs to try and export/import ARAnchors between HoloLens devices. 
It appears that most of the APIs are deprecated and I was able to get export working using a combination of XRAnchorStore, SpatialAnchorManager and the SpatialAnchorExporter. This was a real mess as there wasn't much consistent documentation. Unfortunately, there seemed to be no working method for importing.

I ended up rolling back to an older version of the OpenXR plugin (com.microsoft.mixedreality.openxr-1.8.1) and using the XRAnchorTransferBatch class which contains both Export and Import methods.
